src/Microsoft.Data/DataFrame.cs

```diff
@@ -378,51 +378,62 @@ public DataFrame Sort(string columnName, bool ascending = true)
         }
 
         /// <summary>
-        /// Clips values beyond the specified thresholds
+        /// Clips values beyond the specified thresholds on numeric columns
         /// </summary>
         /// <typeparam name="U"></typeparam>
         /// <param name="lower">Minimum value. All values below this threshold will be set to it</param>
         /// <param name="upper">Maximum value. All values above this threshold will be set to it</param>
-        public DataFrame Clip<U>(U lower, U upper)
+        public DataFrame Clip<U>(U lower, U upper, bool inPlace = false)
         {
-            DataFrame ret = new DataFrame();
-            for (int i = 0; i < ColumnCount; i++)
+            DataFrame ret = this;
+            if (!inPlace)
+                ret = Clone();
+
+            for (int i = 0; i < ret.ColumnCount; i++)
             {
-                BaseColumn column = Column(i);
-                BaseColumn insert = column;
+                BaseColumn column = ret.Column(i);
                 if (column.IsNumericColumn())
-                    insert = column.Clip(lower, upper);
-                ret.InsertColumn(i, insert);
+                    column.Clip(lower, upper, true);
             }
             return ret;
         }
 
         /// <summary>
         /// Adds a prefix to the column names
         /// </summary>
-        public DataFrame AddPrefix(string prefix)
+        public DataFrame AddPrefix(string prefix, bool inPlace = false)
         {
-            for (int i = 0; i < ColumnCount; i++)
+            DataFrame df = this;
+            if (!inPlace)
             {
```

Comment:
It would be nice to consistently brace single line ifs or not. Above you don't use braces, but here you do.

(my personal preference is to always use braces.)